### DISCUSSION TITLE
Backoffice tables lookup in form

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#286] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11.3.0

### Description

Hi guys!

I have a table A in the backoffice showing the following data model:
- name, string
- description, string
- coverages, array of strings

The coverages array of strings contains ids of objects of a collection B.
In the forms that allows me to add a new item in table A, I would like to have the coverages field as a multiselect element that lookups the options from collection B. Is it possible?

Thanks.

**Comment by alexBonadei** [comment, marked as answer]
Yes you can define the format as a multilookup. In the forms you will get a multiselect element for that field.

Here the doc: https://docs.mia-platform.eu/docs/business_suite/backoffice/page_layout#lookups

**Reply by lucascanna** [reply]
Thanks @alexBonadei. I was trying to do it using the Backoffice configurator on Mia-Platform Console.
Do you know if it is possible?

Moreover, I think the documentation refers to data display (e.g. you have the id of an ingredient but you want to show its name that you lookup in another table). I rather need the feature in a data entry form as a multiselect component: the select should show me all the elements of the lookup collection and I select one or more element.

**Comment by clabene** [comment]
@alexBonadei is correct.

Lookup fields are handled differently by different components.
For instance, as you said yourself, the table component will display a label fetched from the remote collection instead of the lookup value.

On the other hand, form components will spawn select fields, allowing to select one or multiple entries - visualized labels fetched from the remote collection (following up on your example, the name of the ingredient). Submitted values will be the corresponding lookup values (the ids of the ingredients). I think this behavior describes your use-case.

Namely, you want the data-schemas of Form and Crud Lookup Client components to configure the `coverages` field like:
```
"coverages": {
  "type": "array",
  "format": "multilookup",
  "lookupOptions": {
    "lookupDataSource": "collection-name",
    "lookupValue": "foreign-key",
    "lookupFields": [
      "label-field", "label-field-2"
    ]
  }
}
```

Nota that for basic pages you usually only create one schema and give it to all components, but this is not required - different components can be provided with different data-schemas, which determines how each one interacts with provided data.

**Comment by lucascanna** [comment]
@clabene @alexBonadei Thanks for the support. After some attempts I managed to do it!

Since I am using the no-code configurator, the missing part was to add the `bk-crud-lookup-client` to my page configuration.
The only way to do it in the no-code configuration was to add it in the JSON file in the advanced section. After adding this component it worked. 
I think this information should be added to the No-code documentation.
